recipes/liblsl/all/conanfile.py

```diff
@@ -47,54 +37,57 @@ def config_options(self):
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version],
-                  destination=self._source_subfolder, strip_root=True)
+        get(self, **self.conan_data["sources"][self.version],
+            destination=self.source_folder, strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["LSL_BUILD_STATIC"] = not self.options.shared
+        tc.variables["LSL_BUNDLED_BOOST"] = False
+        tc.variables["LSL_BUNDLED_PUGIXML"] = False
+        tc.variables["lslgitrevision"] = "v" + self.version
+        tc.variables["lslgitbranch"] = "master"
```

Comment:
Should these be `cache_variables` or is `variables` right here?

Reply:
Just verified that the settings are applied so `variables` appears to be fine.

Reply:
Most of time `variables` is fine. Need to consider those Cmake options in CMakeLists.txt which are listed with `CACHE` property, or when a CMake option is defined before the `project()`  sentence in CMakelists.txt, then you should use `cache_variables`.
Practically speaking, `variables` injects your definition in the CMakeToolchain file, but `cache_variables` passes `-D<cmake_option>=<value>` to your command line.